sdk/storage/azure-storage-common/src/storage_switch_to_secondary_policy.cpp

```diff
@@ -13,7 +13,7 @@ namespace Azure { namespace Storage { namespace _internal {
       const Azure::Core::Context& ctx) const
   {
     std::shared_ptr<bool> replicaStatus;
-    ctx.TryGetValue<std::shared_ptr<bool>>(SecondaryHostReplicaStatusKey, replicaStatus);
+    ctx.TryGetValue(SecondaryHostReplicaStatusKey, replicaStatus);
```

Comment:
Nice! - so we can get the value to get deducted by compiler. Should we remove the rest of the type specifications?

Reply:
We can, but I will leave that as a separate change, and to each area owner. Some folks like being explicit. I don't mind being explicit either.

Reply:
@Jinming-Hu , in storage code, which do you prefer? Explicitly stating the `<T>` or leaving it out and it being inferred?

Reply:
Leave it out